components/google_drive/actions/add-file-sharing-preference/add-file-sharing-preference.js

```diff
@@ -0,0 +1,104 @@
+const googleDrive = require("../../google_drive.app");
+
+/**
+ * Uses Google Drive API to create a permission for a file. The role granted by
+ * the permission is one of `owner`,`organizer`,`fileOrganizer`,
+ * `writer`,`commenter`, `reader`. See the [Google Drive API Reference for
+ * Permissions](https://bit.ly/2XKKG1X) for more information.
+ */
+module.exports = {
+  key: "google_drive-add-file-sharing-preference",
+  name: "Add File Sharing Preference",
+  description:
+    "Add a [sharing](https://support.google.com/drive/answer/7166529) permission to the sharing preferences of a file and provide a sharing URL",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description:
+        "The drive to use. If not specified, your personal Google Drive will be used. If you are connected with any [Google Shared Drives](https://support.google.com/a/users/answer/9310351), you can select it here.",
+      optional: true,
+    },
+    fileId: {
+      propDefinition: [
+        googleDrive,
+        "fileId",
+        (c) => ({
+          drive: c.drive,
+          baseOpts: {
+            q: "'me' in owners",
+          },
+        }),
+      ],
+      optional: false,
+      description: "The file to share",
+    },
+    role: {
```

Comment:
Maybe you can move this to the app? Looks like a variable that can be used in future